library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -486,6 +480,13 @@ public boolean startDownloadIfNotActive(ExecutorService executor) {
         }
     }
 
+    public void updateStatus(int status) {
+        mStatus = status;
+        downloadStatusContentValues.clear();
+        downloadStatusContentValues.put(Downloads.Impl.COLUMN_STATUS, mStatus);
+        mContext.getContentResolver().update(getAllDownloadsUri(), downloadStatusContentValues, null, null);
```

Comment:
so it puts PENDING into the DB just before the thread starts. When does RUNNING go into the DB?